foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/EnableHabushuBuildCacheMigration.java

```diff
@@ -0,0 +1,116 @@
+package com.boozallen.aissemble.upgrade.migration.v1_7_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringReader;
+
+
+/**
+ * This migration adds a new build directory tag to the build node of the project's pom.xml file.
+ */
+public class EnableHabushuBuildCacheMigration extends AbstractAissembleMigration {
+    private static final String BUILD_NODE_PATH = "/project/build";
+    private static final String PACKAGING_NODE_PATH = "/project/packaging";
+    private static final String BUILD_DIR_NODE_PATH = "/project/build/directory";
+
+    // XSL string to strip extra empty newlines from the migrated file-- negates a yucky side effect of the transformer
+    private static final String xsl =
+            "<xsl:stylesheet version=\"1.0\" xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\">\n" +
+            "  <xsl:output indent=\"yes\"/>\n" +
+            "  <xsl:strip-space elements=\"*\"/>\n" +
```

Comment:
Class I'm referring to is `LocationAwareMavenReader`.  It's probably less needed with such a straight-forward edit, but there's also `PomModifications` for batching up several changes and applying them in a single write loop.

Reply:
Discussed offline-- Next steps will be attempting to use `LocationAwareMavenReader` and `PomModifications` to power both `shouldExecute` and `performMigration`.  Current `XPath` approach works well for the current `shouldExecute`, but norming to a singular approach rather than splitting across functions will keep complexity down.

Reply:
Done